geo/src/algorithm/voronoi_diagram.rs

```diff
@@ -4,39 +4,77 @@ use geo_types::coord;
 
 use crate::{Coord, EuclideanLength, GeoFloat, Line, MultiPoint, Point, Polygon, Rect, Triangle};
 
-use crate::triangulate_delaunay::{
-    DelaunayTriangle, DelaunayTriangulationError, DEFAULT_SUPER_TRIANGLE_EXPANSION,
-};
+use crate::triangulate_delaunay::{DelaunayTriangle, DelaunayTriangulationError};
 use crate::{BoundingRect, TriangulateDelaunay};
 
 type Result<T> = result::Result<T, VoronoiDiagramError>;
 
+const DEFAULT_CLIPPING_MASK_EXPANSION: f64 = 20.;
+
+/// The default value for determining if the slope of a line is
+/// near zero or near infinite.
+///
+/// A near infinite slope is greater than this value.
+///
+/// A near zero slope is less than `1 / DEFAULT_SLOPE_THRESHOLD`.
 pub const DEFAULT_SLOPE_THRESHOLD: f64 = 1e3;
 
+/// The components of a Voronoi Diagram.
 #[derive(Debug, Clone)]
 pub struct VoronoiComponents<T: GeoFloat> {
+    /// The Delaunay Triangles used to construct the Voronoi diagram.
     pub delaunay_triangles: Vec<Triangle<T>>,
+    /// The vertices of the Voronoi diagram.
     pub vertices: Vec<Coord<T>>,
+    /// The lines of the Voronoi diagram.
     pub lines: Vec<Line<T>>,
+    /// This is an index of Delaunay triangles that are / have neighours
+    /// the first value is the index of the triangle and the second value
+    /// is the index of a neighbouring triangle.
+    /// If the triangle does not have a neighbour the value is `None`.
     pub neighbours: Vec<(Option<usize>, Option<usize>)>,
 }
 
+/// The clipping mask used to trim the infinity lines of
+/// the Voronoi diagram.
 pub type ClippingMask<T> = Polygon<T>;
 
+/// Compute the Voronoi diagram for a given set of points.
+/// This method uses the property that Delaunay triangulation
+/// [is a dual graph](https://en.wikipedia.org/wiki/Delaunay_triangulation#Relationship_with_the_Voronoi_diagram)
+/// of the Voronoi diagram.
 pub trait VoronoiDiagram<T: GeoFloat>
 where
     f64: From<T>,
 {
+    /// Compute the Voronoi diagram components with a clipping mask to trim infinity lines.
+    /// If `clipping_mask` is set to None, one will be computed by expanding the geometry's
+    /// bounding box.
+    ///
+    /// The `slope_threshold` is used when creating the infinity lines to determine lines with a slope
+    /// of near zero or near infinity.  If `slope_threshold` is set to None, [`DEFAULT_SLOPE_THRESHOLD`] will
+    /// be used.
+    ///
+    /// # Example
+    /// ```rust
+    /// use geo::{coord, polygon, VoronoiDiagram};
+    /// let poly = polygon![
+    ///     coord!{ x: 10., y: 10.},
+    ///     coord!{ x: 10., y: 20.},
+    ///     coord!{ x: 20., y: 20.},
+    ///     coord!{ x: 20., y: 10.},
+    ///     coord!{ x: 10., y: 0.},
+    ///     coord!{ x: 10., y: 0.},
+    ///     coord!{ x: 0., y: 10.},
+    ///     coord!{ x: 0., y: 20.},
+    /// ];
+    /// let voronoi = poly.compute_voronoi_components(None, None).unwrap();
```

Comment:
the user now has a `voronoi`, but what can they do with it? ;-)

maybe assert how many vertices and lines there are in the result ?